citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -77,6 +86,51 @@ public IAuthorityEntry importAuthority(String uri) throws URISyntaxException, Au
         return null;
     }
 
+    /**
+     * This method returns all the authorities found by searching a source based on
+     * first name and/or last name of authority and excludes the authorities that
+     * are imported by the user in citesphere
+     * 
+     * @throws AuthorityImporterNotFoundException
+     */
+    @Override
+    public AuthoritySearchResult searchAuthorityEntries(IUser user, String firstName, String lastName, String source,
+            int page, int pageSize) throws AuthorityServiceConnectionException, AuthorityImporterNotFoundException {
+
+        AuthorityImporter importer = getAuthorityImporter(source);
+
+        if (importer == null) {
+            throw new AuthorityImporterNotFoundException("Authority importer not found for " + source);
+        }
+
+        AuthoritySearchResult searchResult = importer.searchAuthorities(firstName, lastName, page, pageSize);
+
+        if (searchResult.getFoundAuthorities() != null && searchResult.getFoundAuthorities().size() > 0) {
+            List<String> uriList = authorityRepository
+                    .findByUsernameAndNameContainingAndNameContainingOrderByName(user.getUsername(), firstName,
+                            lastName)
+                    .stream().map(IAuthorityEntry::getUri).collect(Collectors.toList());
+  
+            ListIterator<IAuthorityEntry> iter = searchResult.getFoundAuthorities().listIterator();
+            String uri = "";
+            while (iter.hasNext()) {
+                uri = iter.next().getUri();
+                if (!uri.trim().endsWith("/")) {
+                    if (uriList.contains(uri)) {
+                        iter.remove();
+                        continue;
+                    }
+                    uri = uri + "/";
+                }
+                if (uriList.contains(uri)) {
+                    iter.remove();
+                }
```

Comment:
just have one if that checks for a trailing slash and then afterwards to the contains if. No reason to duplicate code here.